lib/ceedling/configurator_builder.rb

```diff
@@ -13,19 +13,33 @@ class ConfiguratorBuilder
 
   def build_global_constants(config)
     config.each_pair do |key, value|
-      formatted_key = key.to_s.upcase
-      # undefine global constant if it already exists
+      # Convert key names to Ruby constant names
+      # Some key names can be C file names that can include dashes
+      # Upcase the key names to create consitency and Ruby constants by convention
+      # Replace dashes with underscores to match handling of Ruby accessor method names
+      formatted_key = key.to_s.gsub('-','_').upcase
+
+      # Undefine global constant if it already exists
       Object.send(:remove_const, formatted_key.to_sym) if @system_wrapper.constants_include?(formatted_key)
-      # create global constant
+
+      # Create global constant
       Object.module_eval("#{formatted_key} = value")
     end
+
+    # TODO: This wants to go somewhere better
+    Object.module_eval("TOOLS_TEST_ASSEMBLER = {}") if (not config[:test_build_use_assembly]) && !defined?(TOOLS_TEST_ASSEMBLER)
+    Object.module_eval("TOOLS_RELEASE_ASSEMBLER = {}") if (not config[:release_build_use_assembly]) && !defined?(TOOLS_RELEASE_ASSEMBLER)
   end
 
 
   def build_accessor_methods(config, context)
+    # Fill configurator object with accessor methods
     config.each_pair do |key, value|
-      # fill configurator object with accessor methods
-      eval("def #{key.to_s.downcase}() return @project_config_hash[:#{key}] end", context)
+      # Convert key names to Ruby method names
+      # Some key names can be C file names that can include dashes; dashes are not allowed in Ruby method names
+      # Downcase the key names and replace any illegal dashes with legal underscores
+      # Downcased key names create consistency and ensure no method names become Ruby constants by accident
+      eval("def #{key.to_s.gsub('-','_').downcase}() return @project_config_hash[:#{key}] end", context)
```

Comment:
Minor comment: I have zero glue about Ruby, but for such "create a good name from arbitrary infput"-tasks in Python I always go by
````python
"".join([ c if c.isalnum() else "_" for c in s ]
```
so that I do not only need to implement every case. Maybe something like that is also a good solution here?

Reply:
And for the special case of creating a function name, we would need to make sure that the first character is a letter, so, me might need to add a prefix.